src/sinol_make/commands/run/__init__.py

```diff
@@ -368,48 +377,98 @@ def print_view(output_file=None):
 		return program_groups_scores
 
 
-	def validate_subtasks(self):
-		print("Validating subtasks...")
-		if 'subtasks' not in self.config.keys():
-			print(util.bold(util.color_red('Subtasks description not defined in config.yml.')))
-			exit(1)
+	def validate_expected_scores(self):
+		print("Validating expected scores...")
+		if 'sinol_expected_scores' not in self.config.keys():
+			print(util.bold(util.color_yellow('Expected scores description not defined in config.yml. ' \
+				    	'The program will run all files on all tests and will print you the results. ')))
+			suggestions = True
 
-		programs = []
-		for subtask in self.config["subtasks"]:
-			score_checksum = 0
-			for group in  self.config["subtasks"][subtask]["groups"]:
-				if group not in self.scores.keys() or group == 0:
-					print(util.bold(util.color_red('Group %d was not defined.' % group)))
-					exit(1)
+			if self.args.apply_suggestions:
+				print(util.bold(util.color_yellow('Suggestions will be applied.')))
+			else:
+				print(util.bold(util.color_yellow('Use flag --apply_suggestions to apply suggestions.')))
 
-				score_checksum += self.scores[group]
-			score_expected = self.config["subtasks"][subtask]["points"]
-			if score_checksum != score_expected:
-				print(util.bold(util.color_red('Subtask %s will grant %d points (expected %d).' % (subtask, score_checksum, score_expected))))
-				exit(1)
+		if suggestions:
+			programs = self.get_solutions()
+		else:
+			for prog in self.config["sinol_expected_scores"]:
+				score_checksum = 0
+				for group, expected_result in self.config["sinol_expected_scores"][prog]["expected"].items():
+					if group not in self.scores.keys() or group == 0:
+						print(util.bold(util.color_red('Group %d was not defined.' % group)))
+						exit(1)
+					if expected_result not in ["TLE", "MLE", "RTE", "WA", "OK"]:
+						print(util.bold(util.color_red('Expected result for group %d is not valid.' % group)))
+						exit(1)
+
+					if expected_result == "OK":
+						score_checksum += self.scores[group]
+
+				score_expected = self.config["sinol_expected_scores"][prog]["points"]
```

Comment:
Can you write me an example of a `sinol_expected_scores`, so that I can confirm that the code does what it should, instead of guessing what it does?

Reply:
Let’s say that there are 3 groups of tests

```
sinol_expected_scores:
  very_cool_name1:
    program: abc1.cop
    points: 33
    expected: {1: OK, 2: WA, 3: ML}
  very_cool_name2:
    program: abc2.cop
    points: 67
    expected: {1: TL, 2: OK, 3: OK}
```

Reply:
We can make the following, which should be cleaner:

```
sinol_expected_scores:
  abc1.cpp:
    points: 33
    expected: {1: OK, 2: WA, 3: ML}
  abc2.cpp:
    points: 67
    expected: {1: TL, 2: OK, 3: OK}
```

Reply:
Sure

Reply:
I changed it.